docs/sphinx/user_guide/feature/resource.rst

```diff
@@ -0,0 +1,247 @@
+.. ##
+.. ## Copyright (c) 2016-20, Lawrence Livermore National Security, LLC
+.. ## and other RAJA project contributors. See the RAJA/COPYRIGHT file
+.. ## for details.
+.. ##
+.. ## SPDX-License-Identifier: (BSD-3-Clause)
+.. ##
+
+.. _resource-label:
+
+=========
+Resources
+=========
+
+This section describes the basic concepts of Resource types and their functionality in ``RAJA::forall``. Resources are used as an interface to various backend constructs and their respective hardware. Currently there exists Resource types for ``Cuda``, ``Hip``, ``OpenMP-target`` and ``Host``. Resource objects allow the user to execute ``RAJA::forall`` calls asynchronously on a respective thread/stream. The underlying concept of each individual Resource is still under development and it should be considered that functionality / behaviour may change.
+
+.. note:: * Currently feature complete asynchronous behaviour and streamed/threaded support is
+            only available when using ``Cuda`` or ``Hip`` resources. 
+          * The ``RAJA::resources`` namespace aliases the ``camp::resources`` namespace.
+
+Each resource has a set of underlying functionality that is synonymous across all resource types.  
+
+ ===================== ===============================================
+ Methods               Brief description
+ ===================== ===============================================
+ get_platform          Returns the underlying camp platform
+                       the resource is associated with.
+ get_event             Return an Event object for the resource from
+                       the last resource call.
+ allocate              Allocate data per the resource's given
+                       backend.
+ deallocate            Deallocate data per the resource's given
+                       backend.
+ memcpy                Perform a memory copy from a src location
+                       to a destination location from the
+                       resource's backend.
+ memset                Set memory value per the resourse's
+                       given backend.
+ wait_for              Enqueue a wait on the resource's stream/thread
+                       for a user passed event to occur.
+ ===================== ===============================================
+  
+Each resource type also defines specific backend information/functionality. For example, each
+Cuda resource contains a ``cudaStream_t`` value with an associated get method. See the 
+individual functionality for each resource in ``raja/tpl/camp/include/resource/``.
```

Comment:
The resource documentation should probably all be in camp, and just linked from RAJA docs? @rhornung67 @trws

Reply:
Agreed. @trws does camp have such documentation to link to? If not, probably keep it in the RAJA user guide for now.

Reply:
It needs to, but isn't in the best position for it.  I have it generating docs through exhale, but don't have a good frame up for the rest.  It's on me right now to get that in better shape.